pkg/auth/authproviders/backend.go

```diff
@@ -11,9 +11,10 @@ import (
 
 // AuthResponse is the response by an auth provider backend that leads to a token issuance.
 type AuthResponse struct {
-	Claims     *tokens.ExternalUserClaim
-	Expiration time.Time
-	ExtraOpts  []tokens.Option
+	Claims          *tokens.ExternalUserClaim
+	Expiration      time.Time
+	ExtraOpts       []tokens.Option
+	UnderlyingToken string
```

Comment:
Maybe name this `IdPToken` or `IDToken`; this seems more generic?

Also, a comment that this is only set for a subset of auth provider backends (i.e. OIDC) would be good.